### ISSUE TITLE
DeepLinkActivity not generated

### ISSUE THREAD

**Issue by athornz** [airbnb/DeepLinkDispatch#24]
I'm having some issues implementing this in my build. I've one successful build where DeepLinkActivity was generated, but most of the time it is not getting generated. 

I have a few other apt libraries in my build so I wonder if that is causing conflicts?

Is there a way I can debug why DeepLinkActivity is not getting generated?

**Comment by felipecsl** [comment]
I've seen issues associated with the gradle daemon causing problems to annotation processing, so if it's failing I'd recommend stopping it `./gradlew --stop` and doing a clean build ``./gradlew clean assemble`.
This fixes problems for me 100% of the time. If this still doesn't work for you, please paste your stack trace here with the exact exception.

**Comment by Denvar** [comment]
I'm having this issue when using multiple DeepLinks into a single Activity.
Compiling & Running works fine with either of these two DeepLiinks on their own, just not when combined.

AndroidManfiest.xml
```xml
<activity
    android:name="com.airbnb.deeplinkdispatch.DeepLinkActivity"
    android:theme="@android:style/Theme.NoDisplay">
    <intent-filter>
        <action android:name="android.intent.action.VIEW" />
        <category android:name="android.intent.category.DEFAULT" />
        <category android:name="android.intent.category.BROWSABLE" />
        <data android:scheme="http"
            android:host="mydomain.com"/>
        <data android:scheme="mydomain"
            android:host="deeplink"/>
    </intent-filter>
</activity>
```

MyActivity.java
```java
@DeepLinks({"http://mydomain.com/deeplink/{id}", "mydomain://deeplink"})
public class MyActivity  extends Activity {

}
```

Exception
```
07-13 18:36:49.285    1896-1896/com.sample.deeplinksample E/AndroidRuntime﹕ FATAL EXCEPTION: main
    Process: com.sample.deeplinksample, PID: 1896
    java.lang.RuntimeException: Unable to instantiate activity ComponentInfo{com.sample.deeplinksample/com.airbnb.deeplinkdispatch.DeepLinkActivity}: java.lang.ClassNotFoundException: Didn't find class "com.airbnb.deeplinkdispatch.DeepLinkActivity" on path: DexPathList[[zip file "/data/app/com.sample.deeplinksample-1/base.apk"],nativeLibraryDirectories=[/vendor/lib, /system/lib]]
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2209)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2360)
            at android.app.ActivityThread.access$800(ActivityThread.java:144)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1278)
            at android.os.Handler.dispatchMessage(Handler.java:102)
            at android.os.Looper.loop(Looper.java:135)
            at android.app.ActivityThread.main(ActivityThread.java:5221)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:899)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:694)
     Caused by: java.lang.ClassNotFoundException: Didn't find class "com.airbnb.deeplinkdispatch.DeepLinkActivity" on path: DexPathList[[zip file "/data/app/com.sample.deeplinksample-1/base.apk"],nativeLibraryDirectories=[/vendor/lib, /system/lib]]
            at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
            at java.lang.ClassLoader.loadClass(ClassLoader.java:511)
            at java.lang.ClassLoader.loadClass(ClassLoader.java:469)
            at android.app.Instrumentation.newActivity(Instrumentation.java:1065)
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2199)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2360)
            at android.app.ActivityThread.access$800(ActivityThread.java:144)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1278)
            at android.os.Handler.dispatchMessage(Handler.java:102)
            at android.os.Looper.loop(Looper.java:135)
            at android.app.ActivityThread.main(ActivityThread.java:5221)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:899)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:694)
    Suppressed: java.lang.ClassNotFoundException: com.airbnb.deeplinkdispatch.DeepLinkActivity
            at java.lang.Class.classForName(Native Method)
            at java.lang.BootClassLoader.findClass(ClassLoader.java:781)
            at java.lang.BootClassLoader.loadClass(ClassLoader.java:841)
            at java.lang.ClassLoader.loadClass(ClassLoader.java:504)
            ... 13 more
     Caused by: java.lang.NoClassDefFoundError: Class not found using the boot class loader; no stack available
```

**Comment by PattaFeuFeu** [comment]
Unfortunately, I still have the same problem: (package/app name changed)

```bash
Process: com.foo.bar.app, PID: 10121
    java.lang.RuntimeException: Unable to instantiate activity ComponentInfo{com.foo.bar.app/com.airbnb.deeplinkdispatch.DeepLinkActivity}: java.lang.ClassNotFoundException: Didn't find class "com.airbnb.deeplinkdispatch.DeepLinkActivity" on path: DexPathList[[zip file "/data/app/com.foo.bar.app-1/base.apk"],nativeLibraryDirectories=[/data/app/com.foo.bar.app-1/lib/arm, /vendor/lib, /system/lib]]
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2328)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2477)
            at android.app.ActivityThread.-wrap11(ActivityThread.java)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1345)
            at android.os.Handler.dispatchMessage(Handler.java:102)
            at android.os.Looper.loop(Looper.java:148)
            at android.app.ActivityThread.main(ActivityThread.java:5415)
            at java.lang.reflect.Method.invoke(Native Method)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:725)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:615)
     Caused by: java.lang.ClassNotFoundException: Didn't find class "com.airbnb.deeplinkdispatch.DeepLinkActivity" on path: DexPathList[[zip file "/data/app/com.foo.bar.app-1/base.apk"],nativeLibraryDirectories=[/data/app/com.foo.bar.app-1/lib/arm, /vendor/lib, /system/lib]]
            at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
            at java.lang.ClassLoader.loadClass(ClassLoader.java:511)
            at java.lang.ClassLoader.loadClass(ClassLoader.java:469)
            at android.app.Instrumentation.newActivity(Instrumentation.java:1067)
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2318)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2477)
            at android.app.ActivityThread.-wrap11(ActivityThread.java)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1345)
            at android.os.Handler.dispatchMessage(Handler.java:102)
            at android.os.Looper.loop(Looper.java:148)
            at android.app.ActivityThread.main(ActivityThread.java:5415)
            at java.lang.reflect.Method.invoke(Native Method)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:725)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:615)
    Suppressed: java.lang.ClassNotFoundException: com.airbnb.deeplinkdispatch.DeepLinkActivity
            at java.lang.Class.classForName(Native Method)
            at java.lang.BootClassLoader.findClass(ClassLoader.java:781)
            at java.lang.BootClassLoader.loadClass(ClassLoader.java:841)
            at java.lang.ClassLoader.loadClass(ClassLoader.java:504)
            ... 12 more
     Caused by: java.lang.NoClassDefFoundError: Class not found using the boot class loader; no stack trace available
```
I did what is described in *Usage* and added a `@DeepLinks` annotation to one of my activities. The  activity part in the AndroidManifest has DeepLinkActivity shown in red as the symbol could not be resolved (which fits with the exception being thrown).

I tried what you suggested (gradlew --stop, gradlew clean assemble) but that didn't change the behaviour.

**Comment by Denvar** [comment]
@PattaFeuFeu be sure you're using 1.3.0-SNAPSHOT as a new release hasn't been made with the fix.
Note I'm still having an issue with this (need to do more testing) but the latest SNAPSHOT release does fix the ClassNotFoundException for me.

**Comment by felipecsl** [comment]
@Denvar is right. This fix is only included in the latest snapshot release (1.3.0-SNAPSHOT) for now

**Comment by alorma** [comment]
I tried to use

```
    compile 'com.airbnb:deeplinkdispatch:1.4.0-SNAPSHOT'
    apt 'com.airbnb:deeplinkdispatch-processor:1.4.0-SNAPSHOT'
```

With no luck
